### ISSUE TITLE
Capture output for IPython Notebook

### ISSUE THREAD

**Issue by wbinventor** [mit-crpg/openmc#390]
The new Python API makes it simple and straightforward to interactively build a model, run a simulation, and process tally data all within a single [IPython Notebook](http://ipython.org/notebook.html). However, the log messages that OpenMC prints to the shell are not captured and displayed by IPython. For example, the following snippet illustrates how one might run an OpenMC simulation from within IPython Notebook using the ``openmc.Executor`` class:

```python
import openmc

# Build a model - settings.xml, geometry.xml, etc.
...

# Use IPython Notebook magic command to capture OpenMC's log messsages
%%capture captured

# Run OpenMC with MPI
executor = openmc.Executor()
executor.run_simulation(mpi_procs=4)

# FIXME: The following should report the stdout stream from OpenMC
# but does not work right now
captured()
```

Unfortunately, the OpenMC output is reported to the console from which the IPython Notebook was launched but *not* within IPython Notebook itself. Perhaps this is a very easy fix, but I've been unable to find any references online that deal with this.

**Comment by paulromano** [comment]
To me the real question is -- is the %%capture cell magic intended to capture output from child processes spawned from the original ipython process? If the answer is no, you'd probably have to change openmc.executor to explicitly capture the output using `subprocess.Popen.communicate` and have it re-print it.

**Comment by wbinventor** [comment]
Thanks for the pointer @paulromano - turns out that it was a pretty easy mod to capture the output as you thought.

**Comment by paulromano** [comment]
Closed via #391
